starter-code/src/viking.js

```diff
@@ -1,11 +1,97 @@
 // Soldier
-function Soldier() {}
+function Soldier(health, strength) {
+  this.health = health;
+  this.strength = strength;
+}
+Soldier.prototype.attack = function(){
+  return this.strength;
+}
+Soldier.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+}
 
 // Viking
-function Viking() {}
+
+function Viking(name, health, strength) {
+  Soldier.call(this, health, strength);
+  this.name = name;
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
+
+
+Viking.prototype.receiveDamage = function(damage){
+  this.health -= damage;
+  if (this.health !== 0) {
+    return this.name + " has received " + damage + " points of damage";
+  } else {
+    return this.name + " has died in act of combat";
+  }
+}
```

Comment:
Haciéndolo así no estáis reutilizando la función `receiveDamage()` de `Soldier`. De esta manera estáis machacando la función original. Lo correcto es modificarla, pero seguir usando la función heredada. Esto se hace llamando a la del padre en la nueva -> `Soldier.prototype.receiveDamage.call(this, damage)` dentro de `Viking.prototype.receiveDamage(damage)`